docs/csharp/tour-of-csharp/classes-and-objects.md

```diff
@@ -0,0 +1,728 @@
+---
+# Classes and objects
+
+***Classes*** are the most fundamental of C#’s types. A class is a data structure that combines state (fields) and actions (methods and other function members) in a single unit. A class provides a definition for dynamically created ***instances*** of the class, also known as ***objects***. Classes support ***inheritance*** and ***polymorphism***, mechanisms whereby ***derived classes*** can extend and specialize ***base classes***.
+
+New classes are created using class declarations. A class declaration starts with a header that specifies the attributes and modifiers of the class, the name of the class, the base class (if given), and the interfaces implemented by the class. The header is followed by the class body, which consists of a list of member declarations written between the delimiters `{` and `}`.
+
+The following is a declaration of a simple class named `Point`:
+
+```csharp
+public class Point
+{
+	public int x, y;
+	public Point(int x, int y) 
+    {
+		this.x = x;
+		this.y = y;
+	}
+}
+```
+
+Instances of classes are created using the new operator, which allocates memory for a new instance, invokes a constructor to initialize the instance, and returns a reference to the instance. The following statements create two Point objects and store references to those objects in two variables:
+
+```csharp
+Point p1 = new Point(0, 0);
+Point p2 = new Point(10, 20);
+```
+The memory occupied by an object is automatically reclaimed when the object is no longer reachable. It is neither necessary nor possible to explicitly deallocate objects in C#.
+
+## Members
+
+The members of a class are either static members or instance members. Static members belong to classes, and instance members belong to objects (instances of classes).
+
+The following provides an overview of the kinds of members a class can contain.
+
+* Constants
+    - Constant values associated with the class
+* Fields
+    - Variables of the class
+* Methods
+    - Computations and actions that can be performed by the class
+* Properties
+    - Actions associated with reading and writing named properties of the class
+* Indexers
+    - Actions associated with indexing instances of the class like an array
+* Events
+    - Notifications that can be generated by the class
+* Operators
+    - Conversions and expression operators supported by the class
+* Constructors
+    - Actions required to initialize instances of the class or the class itself
+* Finalizers
+    - Actions to perform before instances of the class are permanently discarded
+* Types
+    - Nested types declared by the class
+
+## Accessibility
+
+Each member of a class has an associated accessibility, which controls the regions of program text that are able to access the member. There are five possible forms of accessibility. These are summarized below.
+
+* `public`
+	- Access not limited
+* `protected`
+	- Access limited to this class or classes derived from this class
```

Comment:
since there's a not a class or program in discussion, should these descriptions be more generic?